File: kEyB0arddd/sol.js
function sol(str) {
    
    let words = str.replace(/Ctrl \+ (C|X|V)/g, (match, p1) => '#' + p1.toLowerCase()).split(" ");
    let out = "";
    let copied = "";

    for (let i = 0; i < words.length; i++) {
        switch (words[i]) {
            case "#c":
                copied = out;
                break;
            case "#v":
                out += copied;
                break;
            case "#x":
                copied = out.slice(0, i < words.length - 1 ? undefined : -1);
                out = ""
                break;
            default:
                out += words[i] + (i < words.length - 1 ? " " : "");
        }
    }

    return out;
}

//result
let rl = require('readline').createInterface({
    input: process.stdin
});

rl.on("line", (inline) => {
	rl.close();
	console.log(sol(inline));
});

File: kEyB0arddd/sol.py
def sol(s):
    words = s.replace("Ctrl + C", "#c").replace("Ctrl + V", "#v").replace("Ctrl + X", "#x").split()
    out = ""
    copied = ""

    for i in range(len(words)):
        if words[i] == "#c":
            copied = out
        elif words[i] == "#v":
            out += copied
        elif words[i] == "#x":
            copied = out if i < len(words) - 1 else out[:-1]
            out = ""
        else:
            out += words[i] + (" " if i < len(words) - 1 else "")

    return out

# result
inline = input("")
print(sol(inline))
